### PR TITLE
[Magiclysm] Manatouched mutation tree

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Mods "[Magiclysm] Manatouched mutation tree"```
#### Describe alternatives you've considered

Adds a new mutation tree to magiclysm called Manatouched. The idea behind manatouched is that your body is overcharged with magic, making spells much easier to cast than if you were fully human. The main downsides to this mutation tree are being glowy and reduced HP.
#### Testing

Made sure the threshold mutation worked, and checked out the traits that gave spells. Turns out you can't add learned spells to a threshold. I'm actually not sure the glowing is working on these mutations, i copied some stuff from BIOLUM and those were active mutations.
#### Additional context

I'm waiting on #38710 in order to add a few more traits I had in mind. This mutation tree is pretty new, so i expect new traits to be added eventually as well.
image

### CODE REVIEW

File: data/mods/Magiclysm/traits/manatouched.json [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    "threshold": true
+  },
+  {
+    "type": "mutation",
+    "id": "MANA_CRYST_MANA",
+    "name": { "str": "Mana Solidification" },
+    "points": 1,
+    "category": [ "MANATOUCHED" ],
+    "description": "Your mana leylines have become much easier for you to control, resulting in you being able to naturally create crystallized mana.",
+    "spells_learned": [ [ "manatouched_crystal", 5 ] ]
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/traits/manatouched.json]: If I purify this away, do I lose the spell?

Comment (KorGgenT) [reply data/mods/Magiclysm/traits/manatouched.json]: yes, you should.

File: data/mods/Magiclysm/traits/manatouched.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
+[
+  {
+    "type": "mutation_category",
+    "id": "MANATOUCHED",
+    "name": "Manatouched",
+    "threshold_mut": "THRESH_MANA",
+    "mutagen_message": "Your mana leylines surge.",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traits/manatouched.json]:
```suggestion
    "mutagen_message": "Your ley lines surge with mana.",
```

File: data/mods/Magiclysm/traits/manatouched.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    "purifiable": false,
+    "threshold": true
+  },
+  {
+    "type": "mutation",
+    "id": "MANA_CRYST_MANA",
+    "name": { "str": "Mana Solidification" },
+    "points": 1,
+    "category": [ "MANATOUCHED" ],
+    "description": "Your mana leylines have become much easier for you to control, resulting in you being able to naturally create crystallized mana.",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traits/manatouched.json]:
```suggestion
    "description": "Your internal ley lines have become much easier for you to control, resulting in you being able to naturally create crystallized mana.",
```

File: data/mods/Magiclysm/traits/manatouched.json [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    "lumination": [ [ "HAND_R", 8 ], [ "HAND_L", 8 ] ],
+    "prereqs": [ "MANA_LUM" ],
+    "category": [ "MANATOUCHED" ],
+    "spells_learned": [ [ "manatouched_seeker_bolts", 0 ] ]
+  },
+  {
+    "type": "mutation",
+    "id": "MANA_SUBTLE_SPELL",
+    "name": { "str": "Subtle Spell" },
+    "description": "You no longer need to move your arms to cast spells, due to your leylines becoming untethered from your arms.  Arm encumbrance no longer applies to spell failure chance.",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/traits/manatouched.json]:
```suggestion
    "description": "You no longer need to move your arms to cast spells, due to your ley lines becoming untethered from your arms.  Arm encumbrance no longer applies to spell failure chance.",
```

**Pull request merged**